webpack.config.js

```diff
@@ -18,7 +18,11 @@ const generateConfig = ({extensionPath, devMode=false, customOutputPath, analyze
   const aliasesToResolve = {
     "@extensions": '.', /* must provide a default, else it won't compile */
     "@auspice": path.join(__dirname, 'src'),
-    "@libraries": path.join(__dirname, 'node_modules')
+    "@libraries": path.join(__dirname, 'node_modules'),
+    // Pins all react stuff to auspice dir, and uses hot loader's dom (can be used safely in production)
+    "react": path.join(__dirname, 'node_modules/react'), // eslint-disable-line quote-props
```

Comment:
When auspice is a dependency of a project X, then `__dirname` does point to the correct location (`X/node_modules/auspice`) however `react` is not present within ``X/node_modules/auspice/node_modules` for reasons unknown to me. 

I think this approach is the right one in order to ensure that customisations use the same libraries as "core" auspice. My original idea was to force customisations to import code via `import react from "@libraries/react"` but this also had problems (perhaps the same as we're having now).

Reply:
I've had a look and while it feels like a bit of a "nuclear" option, explicitly looking for the "root" node_modules folder in the webpack config seems to do the trick. 

As for the hot loader thing, could you verify if #1126 has the same problem ? I remember resolving this but it was fairly entangled with the react-i18n module IIRC so if it works in #1126 I'd rather not do it twice.

Reply:
Seems by some lucky combination the hot loader actually works well without using the hot-loader dom nor the patch/import, so I pushed a micro commit that reverts those changes and removes the regression.

A more extensive, hopefully more stable solution using the "recommended" implementation of react-hot-loader is found in #1126 and requires a bit of fiddling with react-i18n.

Reply:
#1126 still isn't perfect, but i'd rather pursue the stable solution, even if we do it in a subsequent PR.